llvm/lib/IR/ConstantRange.cpp

```diff
@@ -364,6 +364,29 @@ ConstantRange ConstantRange::makeExactNoWrapRegion(Instruction::BinaryOps BinOp,
   return makeGuaranteedNoWrapRegion(BinOp, ConstantRange(Other), NoWrapKind);
 }
 
+ConstantRange ConstantRange::makeMaskNotEqualRange(const APInt &Mask,
+                                                   const APInt &C) {
+  assert(!Mask.isZero() && "Mask cannot be zero.");
```

Comment:
I feel like this pre-condition should not exist -- we can handle zero mask inside this function.